src/gui/MixerView.cpp

```diff
@@ -68,123 +68,116 @@ MixerView::MixerView() :
+	auto mixerChannelSize = masterView->sizeHint();
 
 	// add mixer channels
-	for( int i = 1; i < m_mixerChannelViews.size(); ++i )
+	for (int i = 1; i < m_mixerChannelViews.size(); ++i)
 	{
 		m_mixerChannelViews[i] = new MixerChannelView(m_channelAreaWidget, this, i);
-		chLayout->addWidget( m_mixerChannelViews[i]->m_mixerLine );
+		chLayout->addWidget(m_mixerChannelViews[i]);
 	}
 
 	// add the scrolling section to the main layout
 	// class solely for scroll area to pass key presses down
 	class ChannelArea : public QScrollArea
 	{
 		public:
-			ChannelArea( QWidget * parent, MixerView * mv ) :
-				QScrollArea( parent ), m_mv( mv ) {}
+			ChannelArea(QWidget * parent, MixerView * mv) :
+				QScrollArea(parent), m_mv(mv) {}
 			~ChannelArea() override = default;
-			void keyPressEvent( QKeyEvent * e ) override
+			void keyPressEvent(QKeyEvent * e) override
 			{
-				m_mv->keyPressEvent( e );
+				m_mv->keyPressEvent(e);
 			}
 		private:
 			MixerView * m_mv;
 	};
-	channelArea = new ChannelArea( this, this );
-	channelArea->setWidget( m_channelAreaWidget );
-	channelArea->setVerticalScrollBarPolicy( Qt::ScrollBarAlwaysOff );
-	channelArea->setFrameStyle( QFrame::NoFrame );
-	channelArea->setMinimumWidth( mixerLineSize.width() * 6 );
-	channelArea->setFixedHeight( mixerLineSize.height() +
-			style()->pixelMetric( QStyle::PM_ScrollBarExtent ) );
-	ml->addWidget( channelArea, 1, Qt::AlignTop );
+	channelArea = new ChannelArea(this, this);
+	channelArea->setWidget(m_channelAreaWidget);
+	channelArea->setVerticalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+	channelArea->setFrameStyle(QFrame::NoFrame);
+	channelArea->setMinimumWidth(mixerChannelSize.width() * 6);
+	channelArea->setFixedHeight(mixerChannelSize.height() +
+			style()->pixelMetric(QStyle::PM_ScrollBarExtent));
+	ml->addWidget(channelArea, 1, Qt::AlignTop);
 
 	// show the add new mixer channel button
 	auto newChannelBtn = new QPushButton(embed::getIconPixmap("new_channel"), QString(), this);
-	newChannelBtn->setObjectName( "newChannelBtn" );
-	newChannelBtn->setFixedSize( mixerLineSize );
-	connect( newChannelBtn, SIGNAL(clicked()), this, SLOT(addNewChannel()));
-	ml->addWidget( newChannelBtn, 0, Qt::AlignTop );
+	newChannelBtn->setObjectName("newChannelBtn");
+	newChannelBtn->setFixedSize(mixerChannelSize);
+	connect(newChannelBtn, SIGNAL(clicked()), this, SLOT(addNewChannel()));
+	ml->addWidget(newChannelBtn, 0, Qt::AlignTop);
 
 
 	// add the stacked layout for the effect racks of mixer channels
-	ml->addWidget( m_racksWidget, 0, Qt::AlignTop | Qt::AlignRight );
+	m_racksWidget->setFixedHeight(mixerChannelSize.height());
+	ml->addWidget(m_racksWidget);
 
-	setCurrentMixerLine( m_mixerChannelViews[0]->m_mixerLine );
+	setCurrentMixerChannel(m_mixerChannelViews[0]);
 
-	setLayout( ml );
 	updateGeometry();
 
 	// timer for updating faders
-	connect( getGUI()->mainWindow(), SIGNAL(periodicUpdate()),
+	connect(getGUI()->mainWindow(), SIGNAL(periodicUpdate()),
```

Comment:
We can combine the line below so it's not split anymore.

Reply:
Could also pass function pointers instead of using `SIGNAL` and `SLOT` macros

Reply:
Fixed with commit 92878ce4030. Using the connect method that takes function pointers now. It's all in one line now.